how-to/how-to-use-virtio-fs-with-kata.md

```diff
@@ -0,0 +1,77 @@
+
+# Kata Containers with virtio-fs
+
+* [Introduction](#introduction)
+* [Pre-requisites](#pre-requisites)
+* [Configure Kata Containers](#configure-kata-containers)
+
+## Introduction
+
+[virtio-fs](https://virtio-fs.gitlab.io/) is a shared file system that lets virtual machines access a directory tree on the host. In Kata, virtio-fs can be used to share the rootfs of the container on the host within the guest to provide significant performance improvements over 9p (the current default sharing mechanism).
+
+**Note: This feature is still experimental and work is being done to get this production ready. Use at your own risk**
+
+This document describes how to get Kata Containers to work with virtio-fs.
+
+## Pre-requisites
+
+This document requires Kata Containers to be [installed](https://github.com/kata-containers/documentation/blob/master/install/README.md) on your system.
+
+* virtio-fs is currently only available with [NEMU](https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-kata-containers-with-nemu.md)
+* This feature currently requires the host to have hugepages support enabled. For eg: this can be enabled with `sysctl vm.nr_hugepages=1024` command on the host. `1024` is the default cache size as defined in kata's configuration file.
+
+## Configure Kata Containers
+
+All you need for this section is to modify the configuration file `/usr/share/defaults/kata-containers/configuration.toml` to specify the options related to this feature.
+
+```diff
+ [hypervisor.qemu]
+-path = "/usr/bin/qemu-lite-system-x86_64"
++path = "/home/foo/build-x86_64_virt/x86_64_virt-softmmu/qemu-system-x86_64_virt"
+ kernel = "/usr/share/kata-containers/vmlinuz.container"
+ initrd = "/usr/share/kata-containers/kata-containers-initrd.img"
+ image = "/usr/share/kata-containers/kata-containers.img"
+-machine_type = "pc"
++machine_type = "virt"
+
+ # Optional space-separated list of options to pass to the guest kernel.
+ # For example, use `kernel_params = "vsyscall=emulate"` if you are having
+@@ -31,7 +31,7 @@ kernel_params = ""
+
+ # Path to the firmware.
+ # If you want that qemu uses the default firmware leave this option empty
+-firmware = ""
++firmware = "/usr/share/nemu/OVMF.fd"
+
+ # Machine accelerators
+ # comma-separated list of machine accelerators to pass to the hypervisor.
+@@ -100,10 +100,10 @@ disable_block_device_use = false
+ # Shared file system type:
+ #   - virtio-9p (default)
+ #   - virtio-fs
+-shared_fs = "virtio-9p"
++shared_fs = "virtio-fs"
+
+ # Path to vhost-user-fs daemon.
+-virtio_fs_daemon = ""
++virtio_fs_daemon = "/home/foo/build-x86_64_virt/virtiofsd-x86_64"
+```
+
+As you can see from the snippet above, all you need to change are:
```

Comment:
Suggested rewrite: 
As you can see from the previous snippet, you only need to change the following: